group-coordinator/src/main/java/org/apache/kafka/coordinator/group/consumer/Assignment.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.coordinator.group.consumer;
+
+import org.apache.kafka.common.Uuid;
+import org.apache.kafka.coordinator.group.generated.ConsumerGroupTargetAssignmentMemberValue;
+
+import java.nio.ByteBuffer;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * An immutable assignment for a member.
+ */
+public class Assignment {
+    public static final Assignment EMPTY = new Assignment(
+        (byte) 0,
+        Collections.emptyMap(),
+        VersionedMetadata.EMPTY
+    );
+
+    /**
+     * The error assigned to the member.
+     */
+    private final byte error;
+
+    /**
+     * The partitions assigned to the member.
+     */
+    private final Map<Uuid, Set<Integer>> partitions;
+
+    /**
+     * The metadata assigned to the member.
+     */
+    private final VersionedMetadata metadata;
+
+    public Assignment(
+        Map<Uuid, Set<Integer>> partitions
+    ) {
+        this(
+            (byte) 0,
+            partitions,
+            VersionedMetadata.EMPTY
+        );
+    }
+
+    public Assignment(
+        byte error,
+        Map<Uuid, Set<Integer>> partitions,
+        VersionedMetadata metadata
+    ) {
+        this.error = error;
+        this.partitions = Collections.unmodifiableMap(Objects.requireNonNull(partitions));
```

Comment:
Are we concerned about the contents of the map being null?

Reply:
I don't think so.